x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/inference_base.ts

```diff
@@ -26,5 +45,26 @@ export abstract class InferenceBase<TInferResponse> {
     this.inputField = model.input?.field_names[0] ?? DEFAULT_INPUT_FIELD;
   }
 
-  protected abstract infer(inputText: string): Promise<TInferResponse>;
+  public setStopped() {
+    this.inferenceError$.next(null);
+    this.runningState$.next(RUNNING_STATE.STOPPED);
+  }
+  public setRunning() {
+    this.inferenceError$.next(null);
+    this.runningState$.next(RUNNING_STATE.RUNNING);
+  }
+
+  public setFinished() {
+    this.runningState$.next(RUNNING_STATE.FINISHED);
+  }
+
+  public setFinishedWithErrors(error: MLHttpFetchError) {
+    this.inferenceError$.next(error);
+    this.runningState$.next(RUNNING_STATE.FINISHED_WITH_ERRORS);
+  }
+
+  protected abstract getInputComponent(): JSX.Element;
+  protected abstract getOutputComponent(): JSX.Element;
+
+  protected abstract infer(): Promise<TInferResponse>;
```

Comment:
Probably one for a follow-up, but shouldn't the `Test model` action be disabled if the model isn't started?

![image](https://user-images.githubusercontent.com/7405507/165922694-5b7ff242-5c2c-4d0f-9d15-ab7c7a6d9437.png)

will give an error `Trained model [bert-base-uncased] is not deployed` when trying to use `Test` on a stopped model.

Reply:
Updated in https://github.com/elastic/kibana/pull/129209/commits/4b033e3f6bb5e5b5610f67f0c5096c4e6ba86286